Library/Homebrew/dev-cmd/update-python-resources.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def update_python_resources_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `update-python-resources` [<options>] <formula>
+
+        Update versions for PyPI resource blocks in <formula>.
+      EOS
+      switch "-p", "--print-only",
+             description: "Print the updated resource blocks instead of changing the formula."
+      named :formula
+    end
+  end
+
+  def update_python_resources
+    update_python_resources_args.parse
+
+    formula = args.formulae.first
+    pypi_name = if formula.stable.url.start_with?("https://files.pythonhosted.org/packages/")
+      File.basename(formula.stable.url).match(/^(.+)-[a-z0-9.]+$/)[1]
+    else
+      formula.name
+    end
+
+    pypi_resources = formula.resources.select do |resource|
+      resource.url.start_with?("https://files.pythonhosted.org/packages/") && resource.name != "homebrew-virtualenv"
+    end
+
+    odie "No PyPI resources found in \"#{formula.name}\"" if pypi_resources.empty?
+
+    odie "\"python@3.8\" must be installed" unless Formula["python@3.8"].any_version_installed?
+
+    odie "\"pipgrip\" must be installed (`pip3 install pipgrip`)" unless File.exist?("#{HOMEBREW_PREFIX}/bin/pipgrip")
+
+    ohai "Retrieving PyPI dependencies for \"#{pypi_name}\"" unless args.print_only?
+    pipgrip_output = JSON.parse(Utils.safe_popen_read("#{HOMEBREW_PREFIX}/bin/pipgrip", "--json", pypi_name))
+
+    ohai "Updating `resource` blocks"
```

Comment:
Should we just emit plain `resource` blocks to stdout, or warn the user somehow if there's a mismatch between the names of the resource blocks in the formula vs what's implied by the dependency tree?

Reply:
I think replacing them might be nice (at least optionally). Having an output-only mode (like `brew bottle`) would be good for testing, too.

Reply:
My ultimate goal is to have it be able to bump resources automatically AND be able to just output the new blocks.

Currently, I have it replace only but I did add a `--print-only` flag with the intention of adding this capability.

Another option, though, is to have this command ONLY output the new blocks and have `bump-formula-pr` or `create` do the replacement after they call this command. Thoughts?

Reply:
Probably want both options since I see three consumers for this command:

1. brew bump-formula-pr
2. brew create
3. user runs it manually

Reply:
I plan on implementing this but haven't done it yet